### ISSUE TITLE
Parameter seperator '+' provided in the LongUrl is getting percent-encoded

### ISSUE THREAD

**Issue by deadmanIsARabbit** [shlinkio/shlink#1075]
#### How Shlink is set-up

* Shlink Version: 2.6.2
* PHP Version: 7.4.17
* How do you serve Shlink: Self-hosted nginx
* Database engine used: MariaDB

#### Summary

LongUrl paramter seperators are getting html encoded. 
'+' results in '%20'

#### Current behavior

In case of shortening an longUrl like 'https://mydomain.com?abc+123' the shortUrl will point to 'https://mydomain.com?abc%20123'

#### Expected behavior

The shortUrl should point back to 'https://mydomain.com?abc+123'

#### How to reproduce
Use the API to shorten any url containing a '+' and try to open the shortUrl

According to https://www.ietf.org/rfc/rfc3986.txt §2.2 the '+' should be valid sub-delim which should be protected from normalization.
Maybe all valid sub-delims shouldn't be encoded by shlink?

Best regards,
Matt

**Comment by acelaya** [comment]
Do you have some public URL including a `+` that I can use to reproduce the issue?

**Comment by deadmanIsARabbit** [comment]
Sorry, non i could post in public.
But i came across another strange behaviour.
Take this one for example:
https://en.wikipedia.org/wiki/2+2_(TV_channel)
shlink produces thist shortUrl: https://cue.watch/ZY3Et
which leads back to 'https://en.wikipedia.org/wiki/2+2_(TV_channel)'

But on the other hand
'https://en.wikipedia.org/wiki/?2+2_(TV_channel)'
produces
https://cue.watch/mncTU
which will take you back to 
https://en.wikipedia.org/wiki/?2%202_%28TV_channel%29

Is this intended and perhaps the correct way to code parameters and only I'm barking up the wrong tree?

**Comment by acelaya** [comment]
I didn't notice your were using `+` in the query string and not the path.

I'm not sure what should be the behavior there. I'll use that wikipedia example URL to investigate a bit further.

**Comment by acelaya** [comment]
Hey @deadmanIsARabbit 

I have been playing around with this, and the encoding on the query params comes from a feature on on Shlink, which makes sure the query string sent to the short URL gets merged with the one hardcoded in the long URL (see https://shlink.io/documentation/some-features/#query-params-forwarding).

In order to do that, they have to be decoded, merged, and encoded again.

However, they both should work the same, as the target server will parse both `+` and `%20` as an empty space.

I have set-up a dummy script which prints the query string, made a request with this query `?withPlus=foo+bar&percentEncoded=foo%20bar`, and I got this result:

![image](https://user-images.githubusercontent.com/2719332/115106382-cbd1c380-9f64-11eb-8564-05783cf7d6b7.png)

So I think I'll keep this behavior as it is. It's quite of an edge case, and there's a risk of introducing a nasty regression if any change is made here.
